spec/acceptance/openvpn_spec.rb

```diff
@@ -2,362 +2,122 @@
-    end
-
-    describe file("#{server_directory}/test_openvpn_server/easy-rsa/vars") do
-      it { is_expected.to be_file }
-      it { is_expected.to contain "export EASY_RSA=\"#{server_directory}/test_openvpn_server/easy-rsa\"" }
-      it { is_expected.to contain '_COUNTRY="CO"' }
-      it { is_expected.to contain '_PROVINCE="ST"' }
-      it { is_expected.to contain '_CITY="A city"' }
-      it { is_expected.to contain '_ORG="FOO"' }
-      it { is_expected.to contain '_EMAIL="bar@foo.org"' }
-    end
-
-    describe file(server_crt.to_s), :crtFile do
-      it { is_expected.to be_file }
-      it { is_expected.to contain 'Issuer: C=CO, ST=ST, L=A city, O=FOO, ' }
-    end
-
-    describe process('openvpn') do
-      it { is_expected.to be_running }
-    end
-
-    describe port(1194) do
-      it { is_expected.to be_listening.with('tcp') }
-    end
-
-    describe command('ip link show tun0') do
-      its(:stdout) { is_expected.to match %r{.* tun0: .*} }
-      its(:exit_status) { is_expected.to eq 0 }
-    end
-
-    describe file("#{key_path}/vpnclienta.key") do
-      it { is_expected.to be_file }
-    end
-
-    describe file("#{crt_path}/vpnclienta.crt") do
-      it { is_expected.to be_file }
-      it { is_expected.to contain 'Issuer: C=CO, ST=ST, L=A city, O=FOO, ' }
+describe 'openvpn', order: :defined do
+  describe 'openvpn::server', order: :defined do
+    describe 'with minimal parameters' do
+      it_behaves_like 'an idempotent resource', 'master' do
+        let(:manifest) do
+          <<-PUPPET
+          openvpn::server { 'test_openvpn_server':
+            country      => 'CO',
+            province     => 'ST',
+            city         => 'A city',
+            organization => 'FOO',
+            email        => 'bar@foo.org',
+            server       => '10.0.0.0 255.255.255.0',
+          }
+          PUPPET
+        end
+      end
+
+      ['/etc/openvpn/server/test_openvpn_server',
+       '/etc/openvpn/server/test_openvpn_server/keys'].each do |dir|
+        describe file(dir) do
+          it { is_expected.to be_directory }
+        end
+
+        describe file('/etc/openvpn/server/test_openvpn_server.conf') do
+          it { is_expected.to be_file }
+        end
+
+        describe service('openvpn-server@test_openvpn_server') do
+          it { is_expected.to be_enabled }
+          it { is_expected.to be_running }
+        end
+
+        describe port(1194) do
+          it { is_expected.to be_listening }
+        end
+
+        describe file('/etc/openvpn/server/test_openvpn_server/easy-rsa/vars') do
+          it { is_expected.to be_file }
+          its(:content) { is_expected.to contain(%r{EASYRSA_REQ_COUNTRY "CO"}) }
+          its(:content) { is_expected.to contain(%r{EASYRSA_REQ_PROVINCE "ST"}) }
+          its(:content) { is_expected.to contain(%r{EASYRSA_REQ_CITY "A city"}) }
+          its(:content) { is_expected.to contain(%r{EASYRSA_REQ_ORG "FOO"}) }
+        end
```

Comment:
These are in a loop but don't use the `dir` variable. They should be outside the `each` loop.